sdk/nodejs/runtime/resource.ts

```diff
@@ -474,16 +475,18 @@ async function resolveOutputs(res: Resource, t: string, name: string,
     }
 
     const label = `resource:${name}[${t}]#...`;
-    for (const key of Object.keys(props)) {
-        if (!allProps.hasOwnProperty(key)) {
-            // input prop the engine didn't give us a final value for.  Just use the value passed into the resource
-            // after round-tripping it through serialization. We do the round-tripping primarily s.t. we ensure that
-            // Output values are handled properly w.r.t. unknowns.
-            const inputProp = await serializeProperty(label, props[key], new Set());
-            if (inputProp === undefined) {
-                continue;
+    if (!isDryRun()) {
+        for (const key of Object.keys(props)) {
+            if (!allProps.hasOwnProperty(key)) {
+                // input prop the engine didn't give us a final value for.  Just use the value passed into the resource
+                // after round-tripping it through serialization. We do the round-tripping primarily s.t. we ensure that
+                // Output values are handled properly w.r.t. unknowns.
+                const inputProp = await serializeProperty(label, props[key], new Set());
+                if (inputProp === undefined) {
+                    continue;
+                }
+                allProps[key] = deserializeProperty(inputProp);
```

Comment:
I'm trying to wrap my head around why we do want to do this for update but don't want to for preview? I'm sure at one point I knew the answer to this - but can't reconstruct in my head right now.

A comment here on the rationale (and links to issues?) may help me and future readers.

Reply:
I also don't understand this comment now:

> We do the round-tripping primarily s.t. we ensure that Output values are handled properly w.r.t. unknowns.

In what cases would there be unknowns here during an update?  Wouldn't all inputs be known before doing the RPC during an update (there can be no computed values running around then, right?)

Reply:
To be honest, I'm not sure that we do want to do this during an update: I left it in for compatibility reasons. We don't want to do this during the preview because we can't know which inputs might or might not be resolved during the update.

Reply:
i'm not certain i get this either.  can you clarify what would happen during preview now?

Reply:
> i'm not certain i get this either. can you clarify what would happen during preview now?

Essentially, we would not run any `apply`s hung off of the outputs of a resource that is to be updated or replaced.